### PR TITLE
Report anonymized data on panic and errors

### PR DESCRIPTION
**Pull request by NetanelBollag**
### Description

Report anonymous data on errors and crashes, only error type and stack trace is sent.

### Testing

This infrastructure change can be tested manually.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR and in github.com/otterize/docs

### CODE REVIEW

File: src/operator/otterizecrds/clientintents-customresourcedefinition.yaml [hunk 1]
```diff
@@ -3,7 +3,6 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.9.0
-  creationTimestamp: null
```
Comment (amitlicht) [inline src/operator/otterizecrds/clientintents-customresourcedefinition.yaml:6]: is this intentional? how is that related to this PR?

File: src/shared/telemetries/telemetriesconfig/config.go [hunk 2]
```diff
@@ -9,8 +9,8 @@ const (
 	TelemetryAPIAddressKey         = "telemetry-address"
 	TimeoutKey                     = "telemetry-client-timeout"
 	CloudClientTimeoutDefault      = "30s"
-	TelemetryEnabledKey            = "telemetry-enabled"
-	TelemetryEnabledDefault        = false
+	TelemetryUsageEnabledKey       = "telemetry-usage-enabled"
```
Comment (amitlicht) [inline src/shared/telemetries/telemetriesconfig/config.go:12]: I'd keep the old flag as is or add a deprecated alias for it to avoid breaking existing deployments, but please double-check with ori (since it's a product definition question)

**Review by amitlicht — APPROVED** [review]
see comments inside

File: src/shared/telemetries/telemetriesconfig/config.go (hunk 2 above)
Comment (NetanelBollag) [reply src/shared/telemetries/telemetriesconfig/config.go:12]: In the helm I will anyway pass `telemetry-errors-enabled` so it will be broken if the image isn't updated. This will be a totally separated chart version since it's a breaking change, so I'm not worry

Comment (NetanelBollag) [reply src/shared/telemetries/telemetriesconfig/config.go:12]: We will keep the old flag as global enabler for the whole
